hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -398,12 +382,10 @@ public Token loadTokenOrFailWith(Id id, ResponseCodeEnum code) {
 	 * to update the {@link com.hedera.services.state.submerkle.ExpirableTxnRecord} of the active transaction
 	 * with these changes.
 	 *
-	 * @param token
-	 * 		the token to save
+	 * @param token the token to save
 	 */
-	public void persistToken(Token token) {
-		final var key = EntityNum.fromLong(token.getId().getNum());
-		final var mutableToken = tokens.get().getForModify(key);
+	public void commitToken(Token token) {
+		final var mutableToken = tokens.getFinalized(token.getId().asGrpcToken());
 		mapModelChangesToMutable(token, mutableToken);
```

Comment:
`mapModelChangesToMutable` should use the `TransactionalLedger` interface and update the properties from there. We should not update the `MerkleToken` class directly, but rather do `tokensLedger.set(tokenId, DELETED, true`) f.e for every property

Reply:
we now load them with getImmutableRef and then do BackingStore.put for the updates